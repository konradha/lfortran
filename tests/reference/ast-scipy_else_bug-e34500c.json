{
    "basename": "ast-scipy_else_bug-e34500c",
    "cmd": "lfortran --indent --fixed-form --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/fixed_form/scipy_else_bug.f",
    "infile_hash": "0e0af6d0d41502c348d645a2260a5a62439a999d046007e7906ebec5",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-scipy_else_bug-e34500c.stdout",
    "stdout_hash": "9ff5418b29bc9b684957c64c619ae61166fd3266109f379c3493028d",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}