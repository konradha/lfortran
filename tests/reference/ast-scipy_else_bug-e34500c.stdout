(TranslationUnit 
    [(Program 
        main 
        () 
        [] 
        [] 
        [] 
        [(If 
            0 
            () 
            (BoolOp 
                (>= n 2) 
                And 
                (<= n (FuncCallOrArray 
                    int 
                    [] 
                    [(() 
                    (* (Real "0.9") x) 
                    ())] 
                    [] 
                    []
                ))
            ) 
            [(= 
                0 
                f0 
                bjv0 
                ()
            )
            (= 
                0 
                f1 
                bjv1 
                ()
            )
            (DoLoop 
                0 
                () 
                45 
                k 
                2 
                n 
                () 
                [(= 
                    0 
                    f 
                    (- (* (/ (* (Real "2.0d0") (- (+ k v0) (Real "1.0d0"))) x) f1) f0) 
                    ()
                )
                (= 
                    0 
                    f0 
                    f1 
                    ()
                )
                (= 
                    0 
                    f1 
                    f 
                    ()
                )
                (= 
                    45 
                    (FuncCallOrArray 
                        vl 
                        [] 
                        [(() 
                        k 
                        ())] 
                        [] 
                        []
                    ) 
                    f 
                    ()
                )] 
                () 
                ()
            )] 
            [(If 
                0 
                () 
                (>= n 2) 
                [(= 
                    0 
                    m 
                    (FuncCallOrArray 
                        msta1 
                        [] 
                        [(() 
                        x 
                        ())
                        (() 
                        200 
                        ())] 
                        [] 
                        []
                    ) 
                    ()
                )
                (If 
                    0 
                    () 
                    (< m n) 
                    [(= 
                        0 
                        n 
                        m 
                        ()
                    )] 
                    [(= 
                        0 
                        m 
                        (FuncCallOrArray 
                            msta2 
                            [] 
                            [(() 
                            x 
                            ())
                            (() 
                            n 
                            ())
                            (() 
                            15 
                            ())] 
                            [] 
                            []
                        ) 
                        ()
                    )] 
                    () 
                    () 
                    ()
                )
                (= 
                    0 
                    f 
                    (Real "0.0d0") 
                    ()
                )
                (= 
                    0 
                    f2 
                    (Real "0.0d0") 
                    ()
                )
                (= 
                    0 
                    f1 
                    (Real "1.0d-100") 
                    ()
                )
                (DoLoop 
                    0 
                    () 
                    50 
                    k 
                    m 
                    0 
                    (u- 1) 
                    [(= 
                        0 
                        f 
                        (- (* (/ (* (Real "2.0d0") (+ (+ v0 k) (Real "1.0d0"))) x) f1) f2) 
                        ()
                    )
                    (If 
                        0 
                        () 
                        (<= k n) 
                        [(= 
                            0 
                            (FuncCallOrArray 
                                vl 
                                [] 
                                [(() 
                                k 
                                ())] 
                                [] 
                                []
                            ) 
                            f 
                            ()
                        )] 
                        [] 
                        () 
                        () 
                        ()
                    )
                    (= 
                        0 
                        f2 
                        f1 
                        ()
                    )
                    (= 
                        50 
                        f1 
                        f 
                        ()
                    )] 
                    () 
                    ()
                )
                (= 
                    0 
                    cs 
                    (Real "0.0d0") 
                    ()
                )
                (If 
                    0 
                    () 
                    (> (FuncCallOrArray 
                        dabs 
                        [] 
                        [(() 
                        bjv0 
                        ())] 
                        [] 
                        []
                    ) (FuncCallOrArray 
                        dabs 
                        [] 
                        [(() 
                        bjv1 
                        ())] 
                        [] 
                        []
                    )) 
                    [(= 
                        0 
                        cs 
                        (/ bjv0 f) 
                        ()
                    )] 
                    [] 
                    () 
                    () 
                    ()
                )
                (= 
                    0 
                    elsecs 
                    (/ bjv1 f2) 
                    ()
                )
                (DoLoop 
                    0 
                    () 
                    55 
                    k 
                    0 
                    n 
                    () 
                    [(= 
                        55 
                        (FuncCallOrArray 
                            vl 
                            [] 
                            [(() 
                            k 
                            ())] 
                            [] 
                            []
                        ) 
                        (* cs (FuncCallOrArray 
                            vl 
                            [] 
                            [(() 
                            k 
                            ())] 
                            [] 
                            []
                        )) 
                        ()
                    )] 
                    () 
                    ()
                )] 
                [] 
                () 
                () 
                ()
            )] 
            () 
            () 
            ()
        )] 
        []
    )]
)
